### PR TITLE
Add spotless for automatic formatting

### PR DESCRIPTION
**Pull request by andreaTP**
Proposal to use an automatic code formatter.

Notes:
- the "eclipse" configuration comes straight [from here](https://github.com/Apicurio/apicurio-studio/blob/main/.config/eclipse/apicurio-eclipse-formatter.xml).
- I'm open to changes to the configuration, just reset to the first commit in this PR, do changes to the config and apply the new style, I'm happy to incorporate the fixes.
- I attempted to prohibit star imports, but it turned out to be more work and changes than expected, descoped
- spotless is now formatting also `xml`, `md` and more files, I think that changes are good but we can exclude things in case is needed.

### CODE REVIEW

**Comment by EricWittmann** [comment]
Note: Waiting on this until after 3.0 Milestone 2 is released.

@andreaTP can you change these two rules:

```
<setting id="org.eclipse.jdt.core.formatter.alignment_for_conditional_expression" value="80"/>
<setting id="org.eclipse.jdt.core.formatter.alignment_for_resources_in_try" value="80"/>
```

Set those to 110 instead of 80?

**Comment by andreaTP** [comment]
@EricWittmann done :+1: 
I put this on hold until we are ready to merge, as everything will break it.
For our future selves:

- pick the spotless configuration in `pom.xml`
- remove the local formatter configuration in `operator/pom.xml`
- pick the `ide-config` folder

do 1 commit with the setup.
Run:
```
mvn spotless:appl
mvn spotless:apply -f operator/pom.xml
mvn spotless:apply examples/pom.xml
```

and you get a "ready to go" PR :slightly_smiling_face:

**Review by EricWittmann — APPROVED** [review]
lgtm

**Pull request merged**
